src/app/courses/courses.component.ts

```diff
@@ -1,12 +1,37 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+
+import { ICourse } from '../shared/models/course.model';
+import { CoursesService } from '../services/courses/courses.service';
 
 @Component({
   selector: 'app-courses',
   templateUrl: './courses.component.html',
   styleUrls: ['./courses.component.scss'],
+  encapsulation: ViewEncapsulation.None,
 })
 export class CoursesComponent implements OnInit {
-  constructor() {}
+  searchQuery = '';
+  courses: ICourse[] = [];
+
+  constructor(private coursesService: CoursesService) {}
+
+  ngOnInit(): void {
+    this.courses = this.coursesService.getCourses();
+  }
+
+  onSearch(): void {
+    console.log(this.searchQuery);
```

Comment:
do we need this console.log?
Please check whole file

Reply:
According to homework we should log search query to console